core/benches/validation.rs

```diff
@@ -111,17 +114,19 @@ fn validate_transaction(criterion: &mut Criterion) {
     .expect("Failed to accept transaction.");
     let mut success_count = 0;
     let mut failures_count = 0;
-    let wsv = build_test_wsv(keys);
     let _ = criterion.bench_function("validate", move |b| {
-        b.iter(|| {
-            match transaction
-                .clone()
-                .validate(&wsv, &AllowAll.into(), &AllowAll.into(), false)
-            {
+        let transaction_validator = TransactionValidator::new(
+            TRANSACTION_LIMITS,
+            Arc::new(AllowAll.into()),
+            Arc::new(AllowAll.into()),
+            Arc::new(build_test_wsv(keys.clone())),
+        );
+        b.iter(
+            || match transaction_validator.validate(transaction.clone(), false) {
```

Comment:
I could do it like that, but it would just forward the call to `transaction_validator` like:
```
impl AcceptedTransaction {
    fn validate(self, validator: TransactionValidator) -> ValidatedTransaction {
        validator.validate(self)
    }
}
```

Reply:
Yes, this is just a preference. Up to you

Reply:
We could follow the `std` library (`From` and `Into`), and create a `Validator`, `Validate`, such that `impl Validate` implies the forwarded `impl Validator`. 

In general, I think that bare associated functions for common functionality should be replaced with `traits`.